### ISSUE TITLE
Matching to register/localize new images into an existing reconstruction

### ISSUE THREAD

**Issue by kondokazuaki** [colmap/colmap#2121]
Hi, thanks for the great tool,

I want to reconstruct camera poses in a 3D model from a video captured by the head mounted camera.
For reducing the time for reconstruction, every 1s images are extracted from the video and used to make 3Dmap.
Then 10fps images are extracted to register them into the reconstructed model to get temporally dense camera poses.
I did 'vocab_tree_matcher' introduced in https://colmap.github.io/faq.html#register-localize-new-images-into-an-existing-reconstruction, but it still need much time (several days) for 2000+ images.

**Describe the bug**
So, I have tried using 'matches_importer' to give image pairs in which each 10fps image corresponds to temporally nearest 1fps image, such as 
10fps_image_01 1fps_image_01
10fps_image_02 1fps_image_01
10fps_image_03 1fps_image_01
.
.
10fps_image_11 1fps_image_02
.
.

Unfortunately, it caused a error : 'ERORR: Image 1fps_image_01 does not exist'.
Of course, '1fps_image_N' were image files used to make 3D map and should be already registered in the model. 

**To Reproduce**
1. make a sparse 3D model with 1fps images
2. do 'feature_extractor' with the reconstructed database and additional 10fps images using '--image_list_path'. it succeeded.
3. do 'matches_importer' using '--match_list_path' containing the above image pair list. 
4. 'Image does not exist' error occurs.

**Expected behavior**
The additional 10fps images are registered to the model via matching with temporally nearest 1fps images that were used to make a 3Dmap.
My question is that using 'vocab_tree_matcher', matching was succeeded between additional 10fps images and 1fps images already registered in the model, but why 'matches_importer' fails it.
Either is great, fixing this error or any other solution for additional image registration within short time :)

**Environment:**
 - OS: Ubuntu 22.04 (colmap was compiled from the source codes)
 - COLMAP Version : latest (3.9?)

**Comment by tsattler** [comment]
> Unfortunately, it caused a error : 'ERORR: Image 1fps_image_01 does not exist'.
Of course, '1fps_image_N' were image files used to make 3D map and should be already registered in the model.

The file name has to include the ending and the file path relative to the image directory specified to the feature extractor.

**Comment by kondokazuaki** [comment]
Thanks for the reply,

If "the ending" means a file extension such as jpg, I'm sorry to forget to include it in the report. 
The filenames were given  as "1fps_image_01.jpg".
And I had also checked the file path problem. 
I modified the source code "feature_matching_utils.cc" to show which images are loaded from database.

> const std::vector<Image> images = database_->ReadAllImages();
>  images_cache_.reserve(images.size());
>  for (const auto& image : images) {
>    //std::cout << "have read : " << image.Name() << std::endl;
>    images_cache_.emplace(image.ImageId(), image);
>  }
 
When "matches_importer" runs, it only showed "10fps_image" and no "1fps_image" was shown even though they were surely used to make 3Dmap before adding 10fps_images. 
I think this may be a reason for the error ...

**Comment by paolovic** [comment]
Hi @tsattler ,

I assume you really busy, so thank you for all the support in all in issues.

Unfortunately, I am having a similar issue and I cannot get it to work.

For now, I want to query a single image against 240 reference images using the vocab tree.
I'm using the python bindings of colmap for convenience, but that shouldn't be the problem.

```py
query_path = "/Users/paolovic/Documents/work/cmp/renders/test"
database_path = "/Users/paolovic/Documents/work/cmp/rooms_samples/colmap_db.db"
pycolmap.extract_features(
        database_path=database_path, image_path=query_path
    )  # This creates a db with just my single query image and it works
    vtmo = pycolmap.VocabTreeMatchingOptions()  # This is needed for vocab_tree matching
    vocab_path = "/Users/paolovic/Documents/work/cmp/vocab_tree_flickr100K_words32K.bin"
    vtmo.vocab_tree_path = vocab_path  # Path to vocab_tree_flickr100K_words32K.bin
    vtmo.match_list_path = "/Users/paolovic/Documents/work/cmp/rooms_samples/rooms_samples.txt" # These are my rooms samples to match against listed in a txt
    result = pycolmap.match_vocabtree(
        database_path=database_path, matching_options=vtmo
    )
```

my rooms_samples.txt looks like this
```
../../rooms_samples/room2/1/00019.jpg
../../rooms_samples/room2/1/00018.jpg
../../rooms_samples/room2/1/00008.jpg
../../rooms_samples/room2/1/00009.jpg
../../rooms_samples/room2/1/00004.jpg
../../rooms_samples/room2/1/00010.jpg
...
```

And it fails with
`Image ../rooms_samples/room7/2/00000.jpg does not exist.`

Am I doing something wrong?
Can you help me out maybe, please?

**Comment by ahojnnes** [comment]
Closing: matches_importer requires all images in same database; workaround provided.
